projects/gateway2/proxy_syncer/proxy_syncer.go

```diff
@@ -588,23 +589,32 @@ func (s *ProxySyncer) Start(ctx context.Context) error {
 				logger.Debug("context done, stopping proxy syncer")
 				return
 			case <-timer.C:
+				// build a list of reports to feed the status plugins
+				var proxiesWithReports []translatorutils.ProxyWithReports
+				var snapPlugins []registry.PluginRegistry
+
 				snaps := s.mostXdsSnapshots.List()
+
 				for _, snapWrap := range snaps {
-					var proxiesWithReports []translatorutils.ProxyWithReports
 					proxiesWithReports = append(proxiesWithReports, snapWrap.proxyWithReport)
+					snapPlugins = append(snapPlugins, snapWrap.pluginRegistry)
 
+					// init the snapshot status plugins
 					initStatusPlugins(ctx, proxiesWithReports, snapWrap.pluginRegistry)
-				}
-				for _, snapWrap := range snaps {
+
 					err := s.proxyTranslator.syncStatus(ctx, snapWrap.proxyKey, snapWrap.fullReports)
```

Comment:
yet another nit: can we add a comment to explain why `syncStatus` here is separate from the plugin based status?

Reply:
Done